File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EPrimaryFixedDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EPrimaryFixedDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd20
            AppTheme.STANDARD -> R.color.cstd20
            AppTheme.BLUE -> R.color.cbd20
            AppTheme.GREEN -> R.color.cgd20
            AppTheme.RED -> R.color.crd20
            AppTheme.NOTHING -> R.color.cnod20
        }
    }
}

File: lib/layer2_core/dark/e_primary_fixed_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EPrimaryFixedDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd20;
      case 'standard': return cstd20;
      case 'blue': return cbd20;
      case 'green': return cgd20;
      case 'red': return crd20;
      case 'nothing': return cnod20;
      default: return cstd20;
    }
  }
}
